internal/pkg/manifest/workload_applyenv_test.go

```diff
@@ -0,0 +1,1846 @@
+				svc.ImageConfig.DependsOn = map[string]string{
+					"mockContainer1": "1",
+					"mockContainer2": "2",
+				}
+				svc.Environments["test"].ImageConfig.Image = Image{}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.DependsOn = map[string]string{
+					"mockContainer1": "1",
+					"mockContainer2": "2",
+				}
+			},
+		},
+		"port overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(1)
+				svc.Environments["test"].ImageConfig.Port = aws.Uint16(2)
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(2)
+			},
+		},
+		"port explicitly overridden by zero value": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(1)
+				svc.Environments["test"].ImageConfig.Port = aws.Uint16(0)
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(0)
+			},
+		},
+		"port not overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(1)
+				svc.Environments["test"].ImageConfig.Image = Image{}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(1)
+			},
+		},
+		"healthcheck overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.HealthCheck = &ContainerHealthCheck{
+					Retries: aws.Int(3),
+				}
+
+				mockInterval1Minute := 60 * time.Second
+				svc.Environments["test"].ImageConfig.HealthCheck = &ContainerHealthCheck{
+					Interval: &mockInterval1Minute,
+					Retries:  aws.Int(5),
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				mockInterval1Minute := 60 * time.Second
+				svc.ImageConfig.HealthCheck = &ContainerHealthCheck{
+					Interval: &mockInterval1Minute,
+					Retries:  aws.Int(5),
+				}
+			},
+		},
+		"healthcheck not overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.HealthCheck = &ContainerHealthCheck{
+					Retries: aws.Int(3),
+				}
+				svc.Environments["test"].ImageConfig.Image = Image{}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.HealthCheck = &ContainerHealthCheck{
+					Command:     nil,
+					Interval:    nil,
+					Retries:     aws.Int(3),
+					Timeout:     nil,
+					StartPeriod: nil,
+				}
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			var inSvc, wantedSvc LoadBalancedWebService
+			inSvc.Environments = map[string]*LoadBalancedWebServiceConfig{
+				"test": {},
+			}
+
+			tc.inSvc(&inSvc)
+			tc.wanted(&wantedSvc)
+
+			got, err := inSvc.ApplyEnv("test")
+
+			require.NoError(t, err)
+			require.Equal(t, &wantedSvc, got)
+		})
+	}
+}
+
+func Test_ApplyEnv_Image_Build(t *testing.T) {
+	testCases := map[string]struct {
+		inSvc  func(svc *LoadBalancedWebService)
+		wanted func(svc *LoadBalancedWebService)
+	}{
+		"composite fields: build string is overridden if build args is not nil": {
```

Comment:
should this be "exclusive fields" to be consistent w the previous test?

Reply:
I meant to differentiate "composite fields" and "exclusive fields". Although in a composite type, the two fields are naturally exclusive, two fields being mutually exclusive doesn't necessarily mean that their parent struct is a composite struct.